service-account/gcp/main.tf

```diff
@@ -0,0 +1,35 @@
+resource "google_service_account" "pods" {
+  account_id  = var.gcp_sa_name
+  project     = var.project_id
+  description = "A gcp service account with the workloadIdentityUser role"
+}
+
+resource "kubernetes_service_account" "pods" {
+  automount_service_account_token = var.automount_service_account_token
+  metadata {
+    name      = var.k8s_sa_name
+    namespace = kubernetes_namespace.pods.metadata[0].name
+    annotations = {
+      "iam.gke.io/gcp-service-account" = google_service_account.pods.email
+    }
+  }
+}
+
+resource "kubernetes_namespace" "pods" {
+  metadata {
+    name = var.K8s_namespace
+  }
+}
+
+resource "google_service_account_iam_member" "pods" {
+  service_account_id = google_service_account.pods.name
+  role               = "roles/iam.workloadIdentityUser"
+  member             = "serviceAccount:${google_service_account.pods.project}.svc.id.goog[${kubernetes_service_account.pods.metadata[0].namespace}/${kubernetes_service_account.pods.metadata[0].name}]"
+}
+
+resource "google_project_iam_member" "workload_identity_sa_bindings" {
```

Comment:
Why do you need project wide permissions? It can be disabled at the organization level, so if you don't actually need it, please remove it.

Reply:
I'm actually granting permissions at the project level because, in this particular case (pubsub service), we don't know the pubsub topics beforehand.